diff --git a/libretro/libretro.c b/libretro/libretro.c
index d9402f0..45a5642 100644
--- a/libretro/libretro.c
+++ b/libretro/libretro.c
@@ -103,7 +103,7 @@ void retro_get_system_info(struct retro_system_info *info)
    memset(info, 0, sizeof(*info));
    info->library_name     = "PrBoom";
    info->library_version  = "v2.5.0";
-   info->need_fullpath    = false;
+   info->need_fullpath    = true;
    info->valid_extensions = "wad|iwad";
    info->block_extract = false;
 }
@@ -277,13 +277,11 @@ bool retro_load_game(const struct retro_game_info *info)
 
    screen_buf = malloc(SURFACE_PIXEL_DEPTH * SCREENWIDTH * SCREENHEIGHT);
 
-   extract_directory(g_wad_dir, info->path, sizeof(g_wad_dir));
-   extract_basename(g_basename, info->path, sizeof(g_basename));
-
-
    argv[argc++] = strdup("prboom");
    if(info->path)
    {
+      extract_directory(g_wad_dir, info->path, sizeof(g_wad_dir));
+      extract_basename(g_basename, info->path, sizeof(g_basename));
       argv[argc++] = strdup("-iwad");
       argv[argc++] = strdup(g_basename);
    }
